include/zephyr/drivers/clock_mgmt/clock.h

```diff
@@ -0,0 +1,428 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * APIs for managing clock objects
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_CLOCK_MGMT_CLOCK_H_
+#define ZEPHYR_INCLUDE_DRIVERS_CLOCK_MGMT_CLOCK_H_
+
+#include <zephyr/devicetree/clock_mgmt.h>
+#include <zephyr/sys/iterable_sections.h>
+#include <zephyr/device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Clock Management Model
+ * @defgroup clock_model Clock device model
+ * @{
+ */
+
+/* Forward declaration of clock driver API struct */
+struct clock_driver_api;
```

Comment:
```suggestion
struct clock_mgmt_driver_api;
```
?

Reply:
Sure, no issue changing the name of this. I'll make the change in the next push (once @XenuIsWatching has a chance to provide feedback on his preference for how to solve the out of tree clock driver header issue)